mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -0,0 +1,251 @@
+
+        :param name: Registered model name.
+        :return: None
+        """
+        req_body = message_to_json(DeleteRegisteredModelRequest(name=name))
+        self._call_endpoint(DeleteRegisteredModelRequest, req_body)
+
+    def search_registered_models(
+        self, filter_string=None, max_results=None, order_by=None, page_token=None
+    ):
+        """
+        Search for registered models in backend that satisfy the filter criteria.
+
+        :param filter_string: Filter query string, defaults to searching all registered models.
+        :param max_results: Maximum number of registered models desired.
+        :param order_by: List of column names with ASC|DESC annotation, to be used for ordering
+                         matching search results.
+        :param page_token: Token specifying the next page of results. It should be obtained from
+                            a ``search_registered_models`` call.
+        :return: A PagedList of :py:class:`mlflow.entities.model_registry.RegisteredModel` objects
+                that satisfy the search expressions. The pagination token for the next page can be
+                obtained via the ``token`` attribute of the object.
+        """
+        _require_arg_unspecified("filter_string", filter_string)
+        _require_arg_unspecified("order_by", order_by)
+        req_body = message_to_json(
+            SearchRegisteredModelsRequest(
+                max_results=max_results,
+                page_token=page_token,
+            )
+        )
+        response_proto = self._call_endpoint(SearchRegisteredModelsRequest, req_body)
+        registered_models = [
+            registered_model_from_uc_proto(registered_model)
+            for registered_model in response_proto.registered_models
+        ]
+        return PagedList(registered_models, response_proto.next_page_token)
+
+    def get_registered_model(self, name):
+        """
+        Get registered model instance by name.
+
+        :param name: Registered model name.
+        :return: A single :py:class:`mlflow.entities.model_registry.RegisteredModel` object.
+        """
+        req_body = message_to_json(GetRegisteredModelRequest(name=name))
+        response_proto = self._call_endpoint(GetRegisteredModelRequest, req_body)
+        return registered_model_from_uc_proto(response_proto.registered_model)
+
+    def get_latest_versions(self, name, stages=None):
+        """
+        Latest version models for each requested stage. If no ``stages`` argument is provided,
+        returns the latest version for each stage.
+
+        :param name: Registered model name.
+        :param stages: List of desired stages. If input list is None, return latest versions for
+                       each stage.
+        :return: List of :py:class:`mlflow.entities.model_registry.ModelVersion` objects.
+        """
+        _raise_unsupported_method(method="get_latest_versions")
```

Comment:
Getting latest version by stage is unsupported for models in UC, as we plan to deprecate stages